diff --git a/quickjs.h b/quickjs.h
--- a/quickjs.h
+++ b/quickjs.h
@@ -222,6 +222,15 @@ typedef struct JSValue {
 #define JS_MKVAL(tag, val) (JSValue){ (JSValueUnion){ .int32 = val }, tag }
 #define JS_MKPTR(tag, p) (JSValue){ (JSValueUnion){ .ptr = p }, tag }
 
+#if defined(__cplusplus)
+#undef JS_MKVAL
+#undef JS_MKPTR
+#define with(T, ...)\
+	([&]{ T t{}; __VA_ARGS__; return t; }())
+#define JS_MKVAL(tag, val) JSValue{ with(JSValueUnion, t.int32 = val), tag }
+#define JS_MKPTR(tag, p) JSValue{ with(JSValueUnion, t.ptr = p), tag }
+#endif
+
 #define JS_TAG_IS_FLOAT64(tag) ((unsigned)(tag) == JS_TAG_FLOAT64)
 
 #define JS_NAN (JSValue){ .u.float64 = JS_FLOAT64_NAN, JS_TAG_FLOAT64 }
