diff --git a/libregexp.c b/libregexp.c
--- a/libregexp.c
+++ b/libregexp.c
@@ -31,6 +31,10 @@
 #include "cutils.h"
 #include "libregexp.h"
 
+#if defined(_WIN32)
+#include <malloc.h>
+#endif
+
 /*
   TODO:
 
diff --git a/quickjs.c b/quickjs.c
--- a/quickjs.c
+++ b/quickjs.c
@@ -45,6 +45,8 @@
 #include <malloc.h>
 #elif defined(__FreeBSD__)
 #include <malloc_np.h>
+#elif defined(_WIN32)
+#include <malloc.h>
 #endif
 
 #include "cutils.h"
