Index: GLee.h
===================================================================
--- GLee.h	(revision 6641)
+++ GLee.h	(revision 6642)
@@ -2261,7 +2261,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glCreateProgram
 #define GLEE_H_DEFINED_glCreateProgram
-  typedef GLuint (APIENTRYP GLEEPFNGLCREATEPROGRAMPROC) ();
+  typedef GLuint (APIENTRYP GLEEPFNGLCREATEPROGRAMPROC) (void);
   GLEE_EXTERN GLEEPFNGLCREATEPROGRAMPROC GLeeFuncPtr_glCreateProgram;
   #define glCreateProgram GLeeFuncPtr_glCreateProgram
 #endif
@@ -2996,7 +2996,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glEndTransformFeedback
 #define GLEE_H_DEFINED_glEndTransformFeedback
-  typedef void (APIENTRYP GLEEPFNGLENDTRANSFORMFEEDBACKPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDTRANSFORMFEEDBACKPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDTRANSFORMFEEDBACKPROC GLeeFuncPtr_glEndTransformFeedback;
   #define glEndTransformFeedback GLeeFuncPtr_glEndTransformFeedback
 #endif
@@ -3038,7 +3038,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glEndConditionalRender
 #define GLEE_H_DEFINED_glEndConditionalRender
-  typedef void (APIENTRYP GLEEPFNGLENDCONDITIONALRENDERPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDCONDITIONALRENDERPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDCONDITIONALRENDERPROC GLeeFuncPtr_glEndConditionalRender;
   #define glEndConditionalRender GLeeFuncPtr_glEndConditionalRender
 #endif
@@ -4808,7 +4808,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glCreateProgramObjectARB
 #define GLEE_H_DEFINED_glCreateProgramObjectARB
-  typedef GLhandleARB (APIENTRYP GLEEPFNGLCREATEPROGRAMOBJECTARBPROC) ();
+  typedef GLhandleARB (APIENTRYP GLEEPFNGLCREATEPROGRAMOBJECTARBPROC) (void);
   GLEE_EXTERN GLEEPFNGLCREATEPROGRAMOBJECTARBPROC GLeeFuncPtr_glCreateProgramObjectARB;
   #define glCreateProgramObjectARB GLeeFuncPtr_glCreateProgramObjectARB
 #endif
@@ -6753,7 +6753,7 @@
 #define GL_INSTRUMENT_MEASUREMENTS_SGIX                    0x8181
 #ifndef GLEE_H_DEFINED_glGetInstrumentsSGIX
 #define GLEE_H_DEFINED_glGetInstrumentsSGIX
-  typedef GLint (APIENTRYP GLEEPFNGLGETINSTRUMENTSSGIXPROC) ();
+  typedef GLint (APIENTRYP GLEEPFNGLGETINSTRUMENTSSGIXPROC) (void);
   GLEE_EXTERN GLEEPFNGLGETINSTRUMENTSSGIXPROC GLeeFuncPtr_glGetInstrumentsSGIX;
   #define glGetInstrumentsSGIX GLeeFuncPtr_glGetInstrumentsSGIX
 #endif
@@ -6777,7 +6777,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glStartInstrumentsSGIX
 #define GLEE_H_DEFINED_glStartInstrumentsSGIX
-  typedef void (APIENTRYP GLEEPFNGLSTARTINSTRUMENTSSGIXPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLSTARTINSTRUMENTSSGIXPROC) (void);
   GLEE_EXTERN GLEEPFNGLSTARTINSTRUMENTSSGIXPROC GLeeFuncPtr_glStartInstrumentsSGIX;
   #define glStartInstrumentsSGIX GLeeFuncPtr_glStartInstrumentsSGIX
 #endif
@@ -6826,7 +6826,7 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_glTagSampleBufferSGIX
 #define GLEE_H_DEFINED_glTagSampleBufferSGIX
-  typedef void (APIENTRYP GLEEPFNGLTAGSAMPLEBUFFERSGIXPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLTAGSAMPLEBUFFERSGIXPROC) (void);
   GLEE_EXTERN GLEEPFNGLTAGSAMPLEBUFFERSGIXPROC GLeeFuncPtr_glTagSampleBufferSGIX;
   #define glTagSampleBufferSGIX GLeeFuncPtr_glTagSampleBufferSGIX
 #endif
@@ -6902,7 +6902,7 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_glFlushRasterSGIX
 #define GLEE_H_DEFINED_glFlushRasterSGIX
-  typedef void (APIENTRYP GLEEPFNGLFLUSHRASTERSGIXPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLFLUSHRASTERSGIXPROC) (void);
   GLEE_EXTERN GLEEPFNGLFLUSHRASTERSGIXPROC GLeeFuncPtr_glFlushRasterSGIX;
   #define glFlushRasterSGIX GLeeFuncPtr_glFlushRasterSGIX
 #endif
@@ -7329,7 +7329,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glUnlockArraysEXT
 #define GLEE_H_DEFINED_glUnlockArraysEXT
-  typedef void (APIENTRYP GLEEPFNGLUNLOCKARRAYSEXTPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLUNLOCKARRAYSEXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLUNLOCKARRAYSEXTPROC GLeeFuncPtr_glUnlockArraysEXT;
   #define glUnlockArraysEXT GLeeFuncPtr_glUnlockArraysEXT
 #endif
@@ -8252,7 +8252,7 @@
 #define GL_TEXTURE_CONSTANT_DATA_SUNX                      0x81D6
 #ifndef GLEE_H_DEFINED_glFinishTextureSUNX
 #define GLEE_H_DEFINED_glFinishTextureSUNX
-  typedef void (APIENTRYP GLEEPFNGLFINISHTEXTURESUNXPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLFINISHTEXTURESUNXPROC) (void);
   GLEE_EXTERN GLEEPFNGLFINISHTEXTURESUNXPROC GLeeFuncPtr_glFinishTextureSUNX;
   #define glFinishTextureSUNX GLeeFuncPtr_glFinishTextureSUNX
 #endif
@@ -8826,7 +8826,7 @@
 #define GL_VERTEX_ARRAY_RANGE_POINTER_NV                   0x8521
 #ifndef GLEE_H_DEFINED_glFlushVertexArrayRangeNV
 #define GLEE_H_DEFINED_glFlushVertexArrayRangeNV
-  typedef void (APIENTRYP GLEEPFNGLFLUSHVERTEXARRAYRANGENVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLFLUSHVERTEXARRAYRANGENVPROC) (void);
   GLEE_EXTERN GLEEPFNGLFLUSHVERTEXARRAYRANGENVPROC GLeeFuncPtr_glFlushVertexArrayRangeNV;
   #define glFlushVertexArrayRangeNV GLeeFuncPtr_glFlushVertexArrayRangeNV
 #endif
@@ -9026,7 +9026,7 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_glResizeBuffersMESA
 #define GLEE_H_DEFINED_glResizeBuffersMESA
-  typedef void (APIENTRYP GLEEPFNGLRESIZEBUFFERSMESAPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLRESIZEBUFFERSMESAPROC) (void);
   GLEE_EXTERN GLEEPFNGLRESIZEBUFFERSMESAPROC GLeeFuncPtr_glResizeBuffersMESA;
   #define glResizeBuffersMESA GLeeFuncPtr_glResizeBuffersMESA
 #endif
@@ -10502,13 +10502,13 @@
 #endif
 #ifndef GLEE_H_DEFINED_glBeginFragmentShaderATI
 #define GLEE_H_DEFINED_glBeginFragmentShaderATI
-  typedef void (APIENTRYP GLEEPFNGLBEGINFRAGMENTSHADERATIPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLBEGINFRAGMENTSHADERATIPROC) (void);
   GLEE_EXTERN GLEEPFNGLBEGINFRAGMENTSHADERATIPROC GLeeFuncPtr_glBeginFragmentShaderATI;
   #define glBeginFragmentShaderATI GLeeFuncPtr_glBeginFragmentShaderATI
 #endif
 #ifndef GLEE_H_DEFINED_glEndFragmentShaderATI
 #define GLEE_H_DEFINED_glEndFragmentShaderATI
-  typedef void (APIENTRYP GLEEPFNGLENDFRAGMENTSHADERATIPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDFRAGMENTSHADERATIPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDFRAGMENTSHADERATIPROC GLeeFuncPtr_glEndFragmentShaderATI;
   #define glEndFragmentShaderATI GLeeFuncPtr_glEndFragmentShaderATI
 #endif
@@ -10803,13 +10803,13 @@
 #define GL_LOCAL_CONSTANT_DATATYPE_EXT                     0x87ED
 #ifndef GLEE_H_DEFINED_glBeginVertexShaderEXT
 #define GLEE_H_DEFINED_glBeginVertexShaderEXT
-  typedef void (APIENTRYP GLEEPFNGLBEGINVERTEXSHADEREXTPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLBEGINVERTEXSHADEREXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLBEGINVERTEXSHADEREXTPROC GLeeFuncPtr_glBeginVertexShaderEXT;
   #define glBeginVertexShaderEXT GLeeFuncPtr_glBeginVertexShaderEXT
 #endif
 #ifndef GLEE_H_DEFINED_glEndVertexShaderEXT
 #define GLEE_H_DEFINED_glEndVertexShaderEXT
-  typedef void (APIENTRYP GLEEPFNGLENDVERTEXSHADEREXTPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDVERTEXSHADEREXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDVERTEXSHADEREXTPROC GLeeFuncPtr_glEndVertexShaderEXT;
   #define glEndVertexShaderEXT GLeeFuncPtr_glEndVertexShaderEXT
 #endif
@@ -11451,7 +11451,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glEndOcclusionQueryNV
 #define GLEE_H_DEFINED_glEndOcclusionQueryNV
-  typedef void (APIENTRYP GLEEPFNGLENDOCCLUSIONQUERYNVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDOCCLUSIONQUERYNVPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDOCCLUSIONQUERYNVPROC GLeeFuncPtr_glEndOcclusionQueryNV;
   #define glEndOcclusionQueryNV GLeeFuncPtr_glEndOcclusionQueryNV
 #endif
@@ -12220,7 +12220,7 @@
 #define GL_PRIMITIVE_RESTART_INDEX_NV                      0x8559
 #ifndef GLEE_H_DEFINED_glPrimitiveRestartNV
 #define GLEE_H_DEFINED_glPrimitiveRestartNV
-  typedef void (APIENTRYP GLEEPFNGLPRIMITIVERESTARTNVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLPRIMITIVERESTARTNVPROC) (void);
   GLEE_EXTERN GLEEPFNGLPRIMITIVERESTARTNVPROC GLeeFuncPtr_glPrimitiveRestartNV;
   #define glPrimitiveRestartNV GLeeFuncPtr_glPrimitiveRestartNV
 #endif
@@ -13160,7 +13160,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glEndTransformFeedbackNV
 #define GLEE_H_DEFINED_glEndTransformFeedbackNV
-  typedef void (APIENTRYP GLEEPFNGLENDTRANSFORMFEEDBACKNVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDTRANSFORMFEEDBACKNVPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDTRANSFORMFEEDBACKNVPROC GLeeFuncPtr_glEndTransformFeedbackNV;
   #define glEndTransformFeedbackNV GLeeFuncPtr_glEndTransformFeedbackNV
 #endif
@@ -13351,7 +13351,7 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_glFrameTerminatorGREMEDY
 #define GLEE_H_DEFINED_glFrameTerminatorGREMEDY
-  typedef void (APIENTRYP GLEEPFNGLFRAMETERMINATORGREMEDYPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLFRAMETERMINATORGREMEDYPROC) (void);
   GLEE_EXTERN GLEEPFNGLFRAMETERMINATORGREMEDYPROC GLeeFuncPtr_glFrameTerminatorGREMEDY;
   #define glFrameTerminatorGREMEDY GLeeFuncPtr_glFrameTerminatorGREMEDY
 #endif
@@ -13375,7 +13375,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glEndConditionalRenderNV
 #define GLEE_H_DEFINED_glEndConditionalRenderNV
-  typedef void (APIENTRYP GLEEPFNGLENDCONDITIONALRENDERNVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDCONDITIONALRENDERNVPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDCONDITIONALRENDERNVPROC GLeeFuncPtr_glEndConditionalRenderNV;
   #define glEndConditionalRenderNV GLeeFuncPtr_glEndConditionalRenderNV
 #endif
@@ -13424,7 +13424,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glEndTransformFeedbackEXT
 #define GLEE_H_DEFINED_glEndTransformFeedbackEXT
-  typedef void (APIENTRYP GLEEPFNGLENDTRANSFORMFEEDBACKEXTPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLENDTRANSFORMFEEDBACKEXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDTRANSFORMFEEDBACKEXTPROC GLeeFuncPtr_glEndTransformFeedbackEXT;
   #define glEndTransformFeedbackEXT GLeeFuncPtr_glEndTransformFeedbackEXT
 #endif
@@ -14680,13 +14680,13 @@
 #endif
 #ifndef GLEE_H_DEFINED_glPauseTransformFeedbackNV
 #define GLEE_H_DEFINED_glPauseTransformFeedbackNV
-  typedef void (APIENTRYP GLEEPFNGLPAUSETRANSFORMFEEDBACKNVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLPAUSETRANSFORMFEEDBACKNVPROC) (void);
   GLEE_EXTERN GLEEPFNGLPAUSETRANSFORMFEEDBACKNVPROC GLeeFuncPtr_glPauseTransformFeedbackNV;
   #define glPauseTransformFeedbackNV GLeeFuncPtr_glPauseTransformFeedbackNV
 #endif
 #ifndef GLEE_H_DEFINED_glResumeTransformFeedbackNV
 #define GLEE_H_DEFINED_glResumeTransformFeedbackNV
-  typedef void (APIENTRYP GLEEPFNGLRESUMETRANSFORMFEEDBACKNVPROC) ();
+  typedef void (APIENTRYP GLEEPFNGLRESUMETRANSFORMFEEDBACKNVPROC) (void);
   GLEE_EXTERN GLEEPFNGLRESUMETRANSFORMFEEDBACKNVPROC GLeeFuncPtr_glResumeTransformFeedbackNV;
   #define glResumeTransformFeedbackNV GLeeFuncPtr_glResumeTransformFeedbackNV
 #endif
@@ -14932,13 +14932,13 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_glBeginSceneEXT
 #define GLEE_H_DEFINED_glBeginSceneEXT
-  typedef GLvoid (APIENTRYP GLEEPFNGLBEGINSCENEEXTPROC) ();
+  typedef GLvoid (APIENTRYP GLEEPFNGLBEGINSCENEEXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLBEGINSCENEEXTPROC GLeeFuncPtr_glBeginSceneEXT;
   #define glBeginSceneEXT GLeeFuncPtr_glBeginSceneEXT
 #endif
 #ifndef GLEE_H_DEFINED_glEndSceneEXT
 #define GLEE_H_DEFINED_glEndSceneEXT
-  typedef GLvoid (APIENTRYP GLEEPFNGLENDSCENEEXTPROC) ();
+  typedef GLvoid (APIENTRYP GLEEPFNGLENDSCENEEXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLENDSCENEEXTPROC GLeeFuncPtr_glEndSceneEXT;
   #define glEndSceneEXT GLeeFuncPtr_glEndSceneEXT
 #endif
@@ -15612,7 +15612,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_wglGetCurrentReadDCARB
 #define GLEE_H_DEFINED_wglGetCurrentReadDCARB
-  typedef HDC (APIENTRYP GLEEPFNWGLGETCURRENTREADDCARBPROC) ();
+  typedef HDC (APIENTRYP GLEEPFNWGLGETCURRENTREADDCARBPROC) (void);
   GLEE_EXTERN GLEEPFNWGLGETCURRENTREADDCARBPROC GLeeFuncPtr_wglGetCurrentReadDCARB;
   #define wglGetCurrentReadDCARB GLeeFuncPtr_wglGetCurrentReadDCARB
 #endif
@@ -15768,7 +15768,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_wglGetCurrentReadDCEXT
 #define GLEE_H_DEFINED_wglGetCurrentReadDCEXT
-  typedef HDC (APIENTRYP GLEEPFNWGLGETCURRENTREADDCEXTPROC) ();
+  typedef HDC (APIENTRYP GLEEPFNWGLGETCURRENTREADDCEXTPROC) (void);
   GLEE_EXTERN GLEEPFNWGLGETCURRENTREADDCEXTPROC GLeeFuncPtr_wglGetCurrentReadDCEXT;
   #define wglGetCurrentReadDCEXT GLeeFuncPtr_wglGetCurrentReadDCEXT
 #endif
@@ -16110,13 +16110,13 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_wglEnableFrameLockI3D
 #define GLEE_H_DEFINED_wglEnableFrameLockI3D
-  typedef BOOL (APIENTRYP GLEEPFNWGLENABLEFRAMELOCKI3DPROC) ();
+  typedef BOOL (APIENTRYP GLEEPFNWGLENABLEFRAMELOCKI3DPROC) (void);
   GLEE_EXTERN GLEEPFNWGLENABLEFRAMELOCKI3DPROC GLeeFuncPtr_wglEnableFrameLockI3D;
   #define wglEnableFrameLockI3D GLeeFuncPtr_wglEnableFrameLockI3D
 #endif
 #ifndef GLEE_H_DEFINED_wglDisableFrameLockI3D
 #define GLEE_H_DEFINED_wglDisableFrameLockI3D
-  typedef BOOL (APIENTRYP GLEEPFNWGLDISABLEFRAMELOCKI3DPROC) ();
+  typedef BOOL (APIENTRYP GLEEPFNWGLDISABLEFRAMELOCKI3DPROC) (void);
   GLEE_EXTERN GLEEPFNWGLDISABLEFRAMELOCKI3DPROC GLeeFuncPtr_wglDisableFrameLockI3D;
   #define wglDisableFrameLockI3D GLeeFuncPtr_wglDisableFrameLockI3D
 #endif
@@ -16365,7 +16365,7 @@
 /* Constants */
 #ifndef GLEE_H_DEFINED_wglGetExtensionsStringEXT
 #define GLEE_H_DEFINED_wglGetExtensionsStringEXT
-  typedef const char * (APIENTRYP GLEEPFNWGLGETEXTENSIONSSTRINGEXTPROC) ();
+  typedef const char * (APIENTRYP GLEEPFNWGLGETEXTENSIONSSTRINGEXTPROC) (void);
   GLEE_EXTERN GLEEPFNWGLGETEXTENSIONSSTRINGEXTPROC GLeeFuncPtr_wglGetExtensionsStringEXT;
   #define wglGetExtensionsStringEXT GLeeFuncPtr_wglGetExtensionsStringEXT
 #endif
@@ -16385,7 +16385,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_wglGetSwapIntervalEXT
 #define GLEE_H_DEFINED_wglGetSwapIntervalEXT
-  typedef int (APIENTRYP GLEEPFNWGLGETSWAPINTERVALEXTPROC) ();
+  typedef int (APIENTRYP GLEEPFNWGLGETSWAPINTERVALEXTPROC) (void);
   GLEE_EXTERN GLEEPFNWGLGETSWAPINTERVALEXTPROC GLeeFuncPtr_wglGetSwapIntervalEXT;
   #define wglGetSwapIntervalEXT GLeeFuncPtr_wglGetSwapIntervalEXT
 #endif
@@ -16469,13 +16469,13 @@
 #endif
 #ifndef GLEE_H_DEFINED_wglBeginFrameTrackingI3D
 #define GLEE_H_DEFINED_wglBeginFrameTrackingI3D
-  typedef BOOL (APIENTRYP GLEEPFNWGLBEGINFRAMETRACKINGI3DPROC) ();
+  typedef BOOL (APIENTRYP GLEEPFNWGLBEGINFRAMETRACKINGI3DPROC) (void);
   GLEE_EXTERN GLEEPFNWGLBEGINFRAMETRACKINGI3DPROC GLeeFuncPtr_wglBeginFrameTrackingI3D;
   #define wglBeginFrameTrackingI3D GLeeFuncPtr_wglBeginFrameTrackingI3D
 #endif
 #ifndef GLEE_H_DEFINED_wglEndFrameTrackingI3D
 #define GLEE_H_DEFINED_wglEndFrameTrackingI3D
-  typedef BOOL (APIENTRYP GLEEPFNWGLENDFRAMETRACKINGI3DPROC) ();
+  typedef BOOL (APIENTRYP GLEEPFNWGLENDFRAMETRACKINGI3DPROC) (void);
   GLEE_EXTERN GLEEPFNWGLENDFRAMETRACKINGI3DPROC GLeeFuncPtr_wglEndFrameTrackingI3D;
   #define wglEndFrameTrackingI3D GLeeFuncPtr_wglEndFrameTrackingI3D
 #endif
@@ -16774,13 +16774,13 @@
 #endif
 #ifndef GLEE_H_DEFINED_glXGetCurrentReadDrawable
 #define GLEE_H_DEFINED_glXGetCurrentReadDrawable
-  typedef GLXDrawable (APIENTRYP GLEEPFNGLXGETCURRENTREADDRAWABLEPROC) ();
+  typedef GLXDrawable (APIENTRYP GLEEPFNGLXGETCURRENTREADDRAWABLEPROC) (void);
   GLEE_EXTERN GLEEPFNGLXGETCURRENTREADDRAWABLEPROC GLeeFuncPtr_glXGetCurrentReadDrawable;
   #define glXGetCurrentReadDrawable GLeeFuncPtr_glXGetCurrentReadDrawable
 #endif
 #ifndef GLEE_H_DEFINED_glXGetCurrentDisplay
 #define GLEE_H_DEFINED_glXGetCurrentDisplay
-  typedef Display * (APIENTRYP GLEEPFNGLXGETCURRENTDISPLAYPROC) ();
+  typedef Display * (APIENTRYP GLEEPFNGLXGETCURRENTDISPLAYPROC) (void);
   GLEE_EXTERN GLEEPFNGLXGETCURRENTDISPLAYPROC GLeeFuncPtr_glXGetCurrentDisplay;
   #define glXGetCurrentDisplay GLeeFuncPtr_glXGetCurrentDisplay
 #endif
@@ -16941,7 +16941,7 @@
 #endif
 #ifndef GLEE_H_DEFINED_glXGetCurrentReadDrawableSGI
 #define GLEE_H_DEFINED_glXGetCurrentReadDrawableSGI
-  typedef GLXDrawable (APIENTRYP GLEEPFNGLXGETCURRENTREADDRAWABLESGIPROC) ();
+  typedef GLXDrawable (APIENTRYP GLEEPFNGLXGETCURRENTREADDRAWABLESGIPROC) (void);
   GLEE_EXTERN GLEEPFNGLXGETCURRENTREADDRAWABLESGIPROC GLeeFuncPtr_glXGetCurrentReadDrawableSGI;
   #define glXGetCurrentReadDrawableSGI GLeeFuncPtr_glXGetCurrentReadDrawableSGI
 #endif
@@ -16969,7 +16969,7 @@
 #define GLX_SCREEN_EXT                                     0x800C
 #ifndef GLEE_H_DEFINED_glXGetCurrentDisplayEXT
 #define GLEE_H_DEFINED_glXGetCurrentDisplayEXT
-  typedef Display * (APIENTRYP GLEEPFNGLXGETCURRENTDISPLAYEXTPROC) ();
+  typedef Display * (APIENTRYP GLEEPFNGLXGETCURRENTDISPLAYEXTPROC) (void);
   GLEE_EXTERN GLEEPFNGLXGETCURRENTDISPLAYEXTPROC GLeeFuncPtr_glXGetCurrentDisplayEXT;
   #define glXGetCurrentDisplayEXT GLeeFuncPtr_glXGetCurrentDisplayEXT
 #endif
