Index: $sourcebase/configure.in
===================================================================
--- $sourcebase/configure.in	(revision 913)
+++ $sourcebase/configure.in	(working copy)
@@ -143,7 +143,7 @@
   FRIBIDI_OBJ="libGLC_la-fribidi.lo libGLC_la-fribidi_char_type.lo \
                libGLC_la-fribidi_mirroring.lo libGLC_la-fribidi_types.lo"
 else
-  PKGCONFIG_REQUIREMENTS="fribidi"
+  PKGCONFIG_REQUIREMENTS=", fribidi"
 fi
 
 # Checks for OpenGL and related libraries.
Index: $sourcebase/quesoglc.pc.in
===================================================================
--- $sourcebase/quesoglc.pc.in	(revision 913)
+++ $sourcebase/quesoglc.pc.in	(working copy)
@@ -9,7 +9,7 @@
 Description: An implementation of SGI's OpenGL Character Renderer (GLC). The OpenGL Character Renderer (GLC) is a state machine that provides OpenGL programs with character rendering services via an application programming interface (API).
 URL: http://quesoglc.sourceforge.net
 Version: @VERSION@
-Requires: fontconfig, freetype2, @PKGCONFIG_REQUIREMENTS@
+Requires: fontconfig, freetype2@PKGCONFIG_REQUIREMENTS@
 Conflicts:
 Libs: -L${libdir} -lGLC
 Libs.private: @PKGCONFIG_LIBS_PRIVATE@

