diff --git a/src/google/protobuf/repeated_ptr_field.h b/src/google/protobuf/repeated_ptr_field.h
index 36ca1c65c..3c3f998a8 100644
--- a/src/google/protobuf/repeated_ptr_field.h
+++ b/src/google/protobuf/repeated_ptr_field.h
@@ -770,7 +770,7 @@ inline void RepeatedPtrFieldBase::MergeFrom<Message>(
 }
 
 // Appends all `std::string` values from `from` to this instance.
-template <>
+template <> PROTOBUF_EXPORT
 void RepeatedPtrFieldBase::MergeFrom<std::string>(
     const RepeatedPtrFieldBase& from);
 
